src/github.com/getlantern/peerscanner/web.go

```diff
@@ -51,36 +56,46 @@ func startHttp() {
 // register is the entry point for peers registering themselves with the service.
 // If peers are successfully vetted, they'll be added to the DNS round robin.
 func register(resp http.ResponseWriter, req *http.Request) {
-	name, ip, port, err := getHostInfo(req)
-	if err == nil && port != 443 {
-		err = fmt.Errorf("Port %d not supported, only port 443 is supported", port)
+	name, ip, port, supportedFronts, err := getHostInfo(req)
+	if err == nil && !(port == "80" || port == "443") {
+		err = fmt.Errorf("Port %d not supported, only ports 80 and 443 are supported", port)
```

Comment:
Yeah the other big factor here is just that we already tunnel HTTPS end-to-end. So the only traffic that's effectively in the clear from CDN -> Lantern (peers or no) is HTTP traffic. Since that's obviously in the clear anyway, I don't think we care at all about the encryption from CDN -> Lantern. In fact, since we're not going to be shipping certs to peer proxies any time soon, there's also no real way we could do that as far as I can tell.